Add support for annotating multiple gem lock files

petervandoros opened the pull request:
It's useful to be able annotate multiple Gemfile lock files in a single PR. It's especially useful when working with a dual booting system such as [bootboot](https://github.com/Shopify/bootboot).

This PR adds support for annotating multiple Gemfile lock files.

orien commented on spec/lib/unwrappr/github/pr_source_spec.rb:
```diff
@@ -19,7 +20,9 @@ module Unwrappr
     let(:lock_file_diff) { double }
```
Is this still needed?

petervandoros replied:
Thanks! Removed in c8a6ad5.

orien commented on CHANGELOG.md:
```diff
@@ -6,7 +6,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
+## [0.6.0] 2021-05-12
+
+### Add
+- Allow specification of Gemfile lock files to annotate. ([#80])
```
```suggestion
- Allow specification of Gemfile lock files to annotate. ([#86])

[0.6.0]: https://github.com/envato/unwrappr/compare/v0.5.0..v0.6.0
[#86]: https://github.com/envato/unwrappr/pull/86
```

petervandoros replied:
Thanks! I forgot to update it after creating the PR 😊 .

orien reviewed (approved):
🎉 Nice one!

orien reviewed (approved):
🚀

johnsyweb commented on lib/unwrappr/cli.rb:
```diff
@@ -17,6 +17,13 @@ class CLI < Clamp::Command
            DESCRIPTION
            attribute_name: :base_branch)
 
+    option ['-f', '--lock-file'],
```
Would this make more sense as a plural?

```suggestion
    option ['-f', '--lock-files'],
```

petervandoros replied:
Given the way to specify multiple gemfile args is to repeat the argument, I chose to keep it singular to indicate that users should provide one file per `--lock-file` arg.

Your suggestion on the usage docs should make that clear to users.

johnsyweb commented on lib/unwrappr/cli.rb:
```diff
@@ -17,6 +17,13 @@ class CLI < Clamp::Command
            attribute_name: :base_branch)
 
+    option ['-f', '--lock-file'],
+           'LOCK_FILE',
```
Let's clarify how to specify multiple filenames:

```suggestion
           'LOCK_FILE1[,LOCK_FILE2][,LOCK_FILE3]',
```

johnsyweb replied:
Or is it 
```suggestion
           LOCK_FILE1 [-f LOCK_FILE2] [-f LOCK_FILE3] [-f ...]',
```

petervandoros replied:
It's the latter.

johnsyweb commented on lib/unwrappr/cli.rb:
```diff
@@ -68,24 +76,24 @@ def execute
+  def self.run_unwapper_in_pwd(base_branch:, lock_files:)
+    return unless lockfiles_present?(lock_files)
 
     puts "Doing the unwrappr thing in #{Dir.pwd}"
```
It would be good to provide some feedback to show that each lockfile has been updated.

petervandoros replied:
Good idea. Added in a0ce2ad.

petervandoros commented on lib/unwrappr/cli.rb:
```diff
@@ -17,6 +17,13 @@ class CLI < Clamp::Command
+           'LOCK_FILE',
+           'The Gemfile.lock files to annotate. Useful when working with multiple lock files.',
+           multivalued: true,
+           default: 'Gemfile.lock',
```
Nice catch. I thought the `multivalued: true` coerced the default value into an Array, but it doesn't!

johnsyweb reviewed (approved):
Thanks for this enhancement. I've made a few UX-based comments but otherwise, I'd be happy to release this.

petervandoros commented:
@johnsyweb @orien thanks for reviewing and providing feedback!

I think I've addressed all the feedback, so would be great if you could cast your eyes over the changes one more time?

johnsyweb reviewed (approved):
Thank you.

petervandoros commented:
@johnsyweb Could you please publish the latest version of this gem when you get a chance?

joesustaric commented:
TIL bootboot.
